crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1281,13 +1281,22 @@ impl PgIndexerStore {
             .map(|deleted_object| deleted_object.clone().into())
             .collect();
         transactional_blocking!(&self.blocking_cp, |conn| {
-            persist_transaction_object_changes(
+            persist_object_mutations(
                 conn,
                 mutated_objects,
-                deleted_objects,
                 object_mutation_latency,
+                object_commit_chunk_counter.clone(),
+            )?;
+            Ok::<(), IndexerError>(())
+        })?;
+        // commit object deletions after mutations b/c objects cannot be mutated after deletion,
+        // otherwise object mutations might override object deletions.
+        transactional_blocking!(&self.blocking_cp, |conn| {
+            persist_object_deletions(
```

Comment:
hmm I think the correct fix should be to ignore the update and only persist the deletion.

Reply:
why is that, for certain object ID, there could be only update and no deletion?

Reply:
@longbowlu @gegaowp just wanted to make sure that we're doing the right thing here. IIUC the current code is technically correct:
1. The object mutation instructions are already filtered to include the last operations only via the `filter_latest_objects()`
2. The object deletion instructions come after and don't require the "latest object" filtering because there can be at most one such deletion: object id is guaranteed to be globally unique.

However, could it be that something like `filter_latest_objects()` over both mutations and deletions would be a cleaner solution? What do you think?